app/models/calendar_event.rb

```diff
@@ -84,8 +84,10 @@ def self.project_date_filter(range, scope = Project)
   end
 
   def self.project_step_date_filter(range, scope = ProjectStep, project_id)
-    scoped_steps = scope.where("scheduled_start_date <= :last AND "\
-      "COALESCE(actual_end_date, scheduled_start_date + scheduled_duration_days) >= :first",
+    scoped_steps = scope.where(
+      "COALESCE(actual_end_date, scheduled_start_date + scheduled_duration_days) >= :first AND "\
+      "scheduled_start_date <= :last OR "\
+      "scheduled_start_date BETWEEN :first AND :last",
```

Comment:
If we just used line 90 and not line 89, would it have the same effect?

Reply:
I could test that out
before we moved around the scopes, one couldn't do without the other

Reply:
Okay. Thanks.

Reply:
@emitche, I've tested it and this OR statement is a necessity